tests/Unit/SerializerTest.php

```diff
@@ -22,6 +22,49 @@ public function testSerializationResult($testCase)
         $this->assertEquals($testCase['expect'], $result);
     }
 
+    public function testSerializationWithMediumSizeArray(): void
+    {
+        $mediumArray = [];
+        $this->fillArray($mediumArray, 1, 20);
+
+        $fixture = new Serializer();
+        $result = $fixture->serializeValue($mediumArray);
+
+        $this->assertEquals('{"1":{"2":{"3":{"4":{"5":{"6":{"7":{"8":{"9":{"10":{"11":{"12":{"13":{"14":{"15":{"16":{"17":{"18":{"19":[]}}}}}}}}}}}}}}}}}}}', $result);
+    }
+
+    public function testSerializationWithLargeArray(): void
+    {
+        // MaxDepth is 1000
+        $largeArray = [];
+        $this->fillArray($largeArray);
+
+        $fixture = new Serializer();
+
+        // json_encode will return false and result is empty string
+        $result = $fixture->serializeValue($largeArray);
+
+        $this->assertEmpty($result);
```

Comment:
its better to check that `$fixture->serializeValue` should return the empty string. Because the method should not return anything else (`null`, `[]`, etc)

```suggestion
        $this->assertEquals($result, '');
```